metadata/src/utils/retain.rs

```diff
@@ -136,38 +157,36 @@ fn update_type(ty: &mut u32, map_ids: &BTreeMap<u32, u32>) {
     *ty = new_id;
 }
 
-/// Strip any pallets out of the RuntimeCall type that aren't the ones we want to keep.
-/// The RuntimeCall type is referenced in a bunch of places, so doing this prevents us from
-/// holding on to stuff in pallets we've asked not to keep.
-fn retain_pallets_in_runtime_call_type<F>(metadata: &mut Metadata, mut filter: F)
+/// Retain the enum type identified by ID and keep only the variants that
+/// match the provided filter.
+fn retain_enum_type<F>(metadata: &mut Metadata, id: u32, mut filter: F)
```

Comment:
Nit; maybe a name like `retain_variants_in_enum_type` would be more descriptive of what it's doing? :)